provider/openstack/cinder.go

```diff
@@ -217,8 +217,13 @@ var _ storage.VolumeSource = (*cinderVolumeSource)(nil)
 func (s *cinderVolumeSource) CreateVolumes(ctx context.ProviderCallContext, args []storage.VolumeParams) ([]storage.CreateVolumesResult, error) {
 	results := make([]storage.CreateVolumesResult, len(args))
 	for i, arg := range args {
-		volume, err := s.createVolume(arg)
+		volume, err := s.createVolume(ctx, arg)
 		if err != nil {
+			_, denied := MaybeHandleCredentialError(err, ctx)
```

Comment:
None of this code is needed - the createVolume() call itself already does the checking for invalid credential. As written, this code will avoid setting the result error if the credential is invalid which in turn will make the caller think the api has succeeded when it hasn't.